src/utils/az.ts

```diff
@@ -36,4 +42,63 @@ export class Az {
 
     return { succeeded: true, result: subs };
   }
+
+  // this method considers getSubscriptions() is called first.
+  async getResourceGroups(subscription: Subscription): Promise<Errorable<ResourceGroup[]>> {
+    if (!(await this.azAccount.waitForLogin())) {
+      commands.executeCommand("azure-account.askForLogin");
+      return { succeeded: false, error: "failed to login" };
+    }
+    await this.azAccount.waitForFilters();
+    const resourceGrps: ResourceGroup[] = [];
+    for (const session of this.azAccount.sessions) {
+      const resources = new ResourceManagementClient(session.credentials2, subscription.subscriptionId!);
+      const resourceGrpPages = resources.resourceGroups.list().byPage();
+      for await (const page of resourceGrpPages) {
+        resourceGrps.push(...page);
+      }
+    }
+    return { succeeded: true, result: resourceGrps };
+  }
+
+  //this method considers getSubscriptions() and getResourceGroups() are called first
+  async getAksClusterNames(subscription: Subscription): Promise<Errorable<ManagedCluster[]>> {
+    const clusters: ManagedCluster[] = [];
+    for (const session of this.azAccount.sessions) {
+      const client = new ContainerServiceClient(session.credentials2, subscription.subscriptionId!);
+      for await (const item of client.managedClusters.list().byPage()) {
+        clusters.push(...item);
+      }
+    }
+    return { succeeded: true, result: clusters };
+  }
+
+  async getAcrRegistries(endpoint: string): Promise<Errorable<Array<string>>> {
```

Comment:
if possible would it be possible to also have a method that just gets the ACR instances belonging to a resource group/subscription? I can totally give it a shot as well. 

The original story's spec for my story won't require us to list out the available repositories/images in the registry, but honestly this can definitely be used anyway! It will definitely allow for a better UX to let users see the images they have in their repo

Reply:
I will check on this, I guess there should be a method for getting the list of ACR for particular resource group. But if not, interim can we ask the user to input the ACR endpoint ex: myacr.acr.io which makes things work?

Reply:
This is completed.